plugin/vim-erlang-tags.vim

```diff
@@ -50,25 +50,66 @@ endfunction
 
 command! ErlangTags call VimErlangTags()
 
-function! VimErlangTagsSelect(split)
-    if a:split
-        split
-    endif
-    let orig_isk = &isk
-    set isk+=:
-    normal "_vawo
-    if getline('.')[col('.') - 2] =~# '[#?]'
-        normal h
+" Execute the given tag lookup for current word, where 'iskeyword' is
+" temporarily set such that modules, records, and macros are included.
+"
+" Accepts a count as an optional second parameter which will be the window
+" height for split or preview commands like 'stag', 'ptjump', etc., or the
+" match index to jump to for 'tag'.
+function! s:GoToErlangTag(cmd, ...)
+    let orig_isk = &l:isk
+    setl isk+=:,#,?
+    let ident = expand('<cword>')
+    let &l:isk = orig_isk
+
+    " With no count arg or zero count, we can execute directly
+    if a:0 == 0 || a:1 == 0
+        execute a:cmd ident
+    else
+        let cnt = a:1
+
+        if a:cmd ==# 'tag'
+            execute cnt a:cmd ident
+        elseif a:cmd =~ '^s'
+            let cmd = 'split +' . a:cmd[1:] . '\ ' . ident
+            execute cnt cmd
+        elseif a:cmd =~ '^p'
+            " :pedit has a cursor movement bug that we work around with a mark
+            " http://bit.ly/1F59kWA  :-(  Affects YouCompleteMe users, for one.
+            normal! m`
+
+            let cmd = 'pedit +' . a:cmd[1:] . '\ ' . ident
+            call s:ExecWithPreviewHeight(cmd, cnt)
+
+            normal! ``
+        endif
     endif
-    let &isk = orig_isk
+endfunction
+
+" Because :pedit can't take height as count like :split does, ugh why.
+function! s:ExecWithPreviewHeight(cmd, height)
+    let orig_height = &l:previewheight
+    let &l:previewheight = a:height
+    execute a:cmd
+    let &l:previewheight = orig_height
```

Comment:
Ah yes, my intuition was coming from somewhere :smile: 

It turns out that the `'previewheight'` value was restored on error (because execution continued, `:h except-compat`), but there is another reason to catch the error: it leaves behind a spurious preview window of the current buffer, as if you'd run `:pedit` with no argument. Fixed.

I can squash these changes later if desired, just going to leave it until further review.